### PR TITLE
Travis CI: Don’t allow bare exceptions

### PR DESCRIPTION
**Pull request by cclauss**
### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
$ `flake8 . --count --select=E101,E722,E9,F4,F63,F7,F82,W191 --show-source --statistics`
```
./linear_algebra/src/test_linear_algebra.py:25:9: E722 do not use bare 'except'
        except:
        ^
./ciphers/xor_cipher.py:151:9: E722 do not use bare 'except'
        except:
        ^
./ciphers/xor_cipher.py:176:9: E722 do not use bare 'except'
        except:
        ^
./hashes/hamming_code.py:128:17: E722 do not use bare 'except'
                except:
                ^
./hashes/hamming_code.py:232:17: E722 do not use bare 'except'
                except:
                ^
5     E722 do not use bare 'except'
5
```

File: hashes/hamming_code.py [hunk 1]
```diff
@@ -229,7 +229,7 @@ def receptorConverter(sizePar, data):
             if x != None:
                 try:
                     aux = (binPos[contLoop])[-1 * (bp)]
-                except:
+                except IOError::
```
Comment (TheSuperNoob) [inline hashes/hamming_code.py]: Whoops, small typo :)

**Comment by TheSuperNoob** [comment]
Also maybe update [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md) to reflect the changes.

**Comment by cclauss** [comment]
Please do not land this PR until the tests pass.  Can someone please help me understand the intended logic of the failing test?!?

**Comment by TheSuperNooby** [comment]
> Please do not land this PR until the tests pass. Can someone please help me understand the intended logic of the failing test?!?
```
$ flake8 . --count --select=E101,E722,E9,F4,F63,F7,F82,W191 --show-source --statistics

./linear_algebra/src/test_linear_algebra.py:25:9: E722 do not use bare 'except'
        except:
        ^$ flake8 . --count --select=E101,E722,E9,F4,F63,F7,F82,W191 --show-source --statistics
```

I think you just forgot to fix this last file

**Comment by cclauss** [comment]
I do not know how to fix the last file because I do not understand it’s intention.

**Comment by TheSuperNoob** [comment]
> I do not know how to fix the last file because I do not understand it’s intention.

You can do `except Exception: ` for all generic exceptions

**Comment by poyea** [comment]
@TheSuperNoob `IndexError` is a kind of `Exception`. Seems that Travis ran the test on the old version of it.

**Comment by TheSuperNoob** [comment]
I'm not seeing any changes to the `./linear_algebra/src/test_linear_algebra.py` file in this PR
@poyea

**Comment by poyea** [comment]
@TheSuperNoob Oh, you're right.

**Comment by cclauss** [comment]
except Exception: is just as bad... https://realpython.com/the-most-diabolical-python-antipattern

**Pull request merged**